src/ol/render/canvas/replaygroup.js

```diff
@@ -109,16 +183,19 @@ ol.render.canvas.ReplayGroup.prototype.finish = function() {
  * @template T
  */
 ol.render.canvas.ReplayGroup.prototype.forEachFeatureAtCoordinate = function(
-    coordinate, resolution, rotation, skippedFeaturesHash, callback) {
+    coordinate, resolution, rotation, hitTolerance, skippedFeaturesHash, callback) {
 
-  var transform = ol.transform.compose(this.hitDetectionTransform_,
-      0.5, 0.5,
+  hitTolerance = Math.round(hitTolerance);
+  var contextSize = hitTolerance * 2 + 1;
+  var transform = ol.transform.compose(ol.transform.create(),
```

Comment:
To produce less garbage, it would be better here to reuse an existing transform, like `this.hitDetectionTransform_` in the current code.